### ISSUE TITLE
Discussion: The Component Management Mess

### ISSUE THREAD

**Issue by bdkjones** [bower/bower#1791]
This issue is a discussion of the current state of how Bower manages component information. I'll illustrate the problem using two examples:
### JQuery Missing

If you go to bower.io/search right now and search for "jQuery", the main jQuery component is missing. You'll get a bunch of plugins and things like jQuery UI, JQuery Color, etc, but the main jQuery component is not returned. I confirmed this by pulling the JSON file that the search routine hits. It's nowhere in that list. It was there earlier, it's not now. To prove this, here's a screenshot of the results: 

![screen shot 2015-04-30 at 22 21 06](https://cloud.githubusercontent.com/assets/926566/7427022/4579c442-ef87-11e4-878d-10de81b88def.png)
### Breakpoint-Sass

If you use the CLI to pull up info about `breakpoint-sass`, you'll see that it lists this as the URL for homepage: 'https://github.com/Team-Sass/breakpoint' That URL is 404-ed. 

When you run `bower install breakpoint-sass`, the CLI is pulling version 2.5.0 of breakpoint-sass using this URI: https://github.com/Team-Sass/breakpoint/archive/v2.5.0.tar.gz

That URI is still valid; if you load it in your browser, it will pull down the package. But the rest of that link is dead and breakpoint-sass has moved to a new home at: https://github.com/at-import/breakpoint

If you attempt to go to the old GitHub page, it redirects to the new one. But the Bower CLI is still pulling from the old, dead repo. The bower.io/search page, however, loads the NEW URL for breakpoint-sass. (Because the search page is pulling from a different JSON list at herokuapp.com than the CLI pulls from). 
### How Can We Fix This?

The examples above are two of about 3,500 that I've identified. (I write CodeKit and I have a script that hits the same JSON list that the CLI uses and retrieves info about each component.) But the CLI list has TONS of incorrect information, such as the outdated links for breakpoint-sass. Many of the repos are 404-ed. There are components literally named `""` and `" "`. (That's a component with no name and a component with a space character for the name.)

It's becoming a bit untenable trying to make sense of the way Bower handles components. The CLI is doing one thing, but the public search website is doing another. There's two separate lists of components for each of these and both have big errors and omissions (the search list currently doesn't have jQuery!)

There's got to be a better way, right? Can we consolidate Bower's sources so that there is **ONE** canonical list of components that is correct? Can we implement a way for component authors to update their information when they, for example, move a repo?

Is there a solution being worked on? If so, can we prioritize that? 

In short: can we get a One Ring To Rule Them All?

**Comment by bdkjones** [comment]
Someone on Twitter pointed out that jQuery is not *really* a Bower package and that if you run `bower install jquery`, it just downloads the jQuery repo. But isn't that sort of semantics? If jQuery is registered and installable with the CLI, shouldn't it also be searchable on the site? 

It just seems to me that consolidating into ONE official list of components is the best solution. If something can be installed with Bower, it ought to be listed in the search results and both places ought to have the same information about that component.

**Comment by bdkjones** [comment]
This appears to be related: https://github.com/bower/registry/issues/112

**Comment by bdkjones** [comment]
I was mistaken. There are 5,723 components registered with Bower and available to install from the CLI that are NOT found on bower.io/search.

**Comment by sheerun** [comment]
This no longer seems to be the case as jquery returns proper result
